source/nielvisiii/academicIO.py

```diff
@@ -425,15 +435,55 @@ def __init__(self, *configuration):
             configuration_details['bank'] = configuration_details['bank'].value
             configuration_details['channel'] = configuration_details['channel'].value
             configuration_details['value_address'] = self.session.registers['AO.' + configuration_details['bank'] + '_' + str(configuration_details['channel']) + '.VAL']
-            self.go = self.session.registers['AO.SYS.GO']
-            self.stat = self.session.registers['AO.SYS.STAT']
-            self.lsb_weights = 4882813.0 / 1E+9
-            self.offset = 0.0 / 1E+9
-            self.signed = True
             self.enamask = configuration_details['channel']^2
             self.channel_list.append(configuration_details)
 
-    def write(self, value):
+        for bank in Bank:
+            self.dma_idl[bank.value] = self.session.registers['AO.%s.DMA_IDL' % bank.value]
+            self.dma_cntr[bank.value] = self.session.registers['AO.%s.DMA_CNTR' % bank.value]
+            self.dma_ena[bank.value] = self.session.registers['AO.%s.DMA_ENA' % bank.value]
+            
+        self.go = self.session.registers['AO.SYS.GO']
+        self.stat = self.session.registers['AO.SYS.STAT']
+        self.lsb_weights = 4882813.0 / 1E+9
+        self.offset = 0.0 / 1E+9
+        self.signed = True
+
+    def write(self, *args):
+        """
+        Writes values to one or more analog output channels. Use the
+        write(value) function to write a single point of data to the channel. Use
+        the write(values, sample_rate) to write multiple points of data to the
+        channel. The function does not return until the write to physical I/O
```

Comment:
function means everything inside `def write():`. The program exists this function after the write completes.